src/pages/NftCollections/index.tsx

```diff
@@ -60,8 +61,11 @@ const NftCollections = () => {
           </a>
         </div>
         <div className={styles.list}>
-          <ListOnDesktop isLoading={isLoading} list={list} />
-          <ListOnMobile isLoading={isLoading} list={list} />
+          {isMobile ? (
+            <ListOnMobile isLoading={isLoading} list={tags === '' ? [] : list} />
```

Comment:
Unnecessary `tags === ''`. If it is necessary, the condition can be prepended in

```js
const list = data?.data ?? []
```

at line 35

Reply:
> Unnecessary `tags === ''`. If it is necessary, the condition can be prepended in
> 
> ```js
> const list = data?.data ?? []
> ```
> 
> at line 35

`undefined` means query data without filter, and the empty string means not to query data, and  list nothing

Reply:
> > Unnecessary `tags === ''`. If it is necessary, the condition can be prepended in
> > ```js
> > const list = data?.data ?? []
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > at line 35
> 
> `undefined` means query data without filter, and the empty string means not to query data, and list nothing

The statement is
```
if tags === '' then
  list is empty
else
  list is fetched from the network
```

That means the network request is unnecessary when `tags === ''`

Reply:
Well, I think it is the same as what I did, if the tags is the empty string, use the empty array, or use the data from the api.

Reply:
Currently, the `list` will be fetched via HTTP request when `tags === ''`, but it's not used.

The steps are

1. fetch the list from backend
2. pass `[]` to the component when `tags === ''`

Step 1 can be avoided when `tags === ''`

Reply:
Any update on this conversation?